### PR TITLE
[LUMI-21] Make required fields as `required` in chart

### PR DESCRIPTION
**Pull request by javiermtorres**
Chart values that are needed for the Deployment env vars should be made `required`, and show an error if they are null-valued (following the Go template definition).

### CODE REVIEW

File: lumigator/infra/mzai/helm/lumigator/templates/deployment.yaml [hunk 1]
```diff
@@ -37,25 +37,25 @@ spec:
               protocol: TCP
           env:
```
Comment (macaab26) [inline lumigator/infra/mzai/helm/lumigator/templates/deployment.yaml:38]: Do you only add it for the env vars? Is there a reason why?

**Comment by chainlink** [comment]
Tag and repository are also required

**Review by chainlink — APPROVED** [review]
Added command on repo and tag, but otherwise pre-approving

**Comment by javiermtorres** [comment]
> Tag and repository are also required

But shouldn't we have default values for these in the `values.yaml` file? The user wouldn't need to add these in their own values.

**Comment by macaab26** [comment]
FYI, I created a [story](https://mzai.atlassian.net/browse/LUMI-52) to check all the default values for the variables, and also check if we miss some required variable.

**Review by macaab26 — APPROVED** [review]

**Review by agpituk — APPROVED** [review]

**Comment by agpituk** [comment]
> > Tag and repository are also required

> But shouldn't we have default values for these in the `values.yaml` file? The user shouldn't need to add these in their own values.

I'm in favor of what Javier says, but, in that case, shouldn't we do add a default for the repository as well?

**Comment by agpituk** [comment]
> > > Tag and repository are also required
> 
> > But shouldn't we have default values for these in the `values.yaml` file? The user shouldn't need to add these in their own values.
> 
> I'm in favor of what Javier says, but, in that case, shouldn't we do add a default for the repository as well?

So I just had a chat with Mario which highlighted a few [things](https://mzlai.slack.com/archives/C07LJP9Q5KJ/p1726217228011579?thread_ts=1726209826.507089&cid=C07LJP9Q5KJ). After that I think required + defaults is the way to go

**Pull request merged**